### PR TITLE
Extract shared jobs AndCultureCode/AndCultureCode.DotnetReact.Boilerplate#4

### PR DESCRIPTION
**Pull request by snsavage**
- [x] Related GitHub issue(s) linked in PR description
- [x] Destination branch merged, built and tested with your changes
- [x] Code formatted and follows best practices and patterns
- [x] Code builds cleanly (no _additional_ warnings or errors)
- [x] Manually tested
- [x] Automated tests are passing
- [ ] No _decreases_ in automated test coverage
- [x] Documentation updated (readme, docs, comments, etc.)
- [ ] Localization: No hard-coded error messages in code files (_minimally_ in string constants)

### CODE REVIEW

File: src/AndcultureCode.CSharp.Core/Interfaces/Conductors/Jobs/IJobEnqueueConductor.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        IResult<Job> Enqueue(Type workerType, List<object> workerArgs, long? startedById);
+
+        /// <summary>
+        /// Validates that a TWorker type/interface only has one implementing class.
+        /// </summary>
+        /// <typeparam name="TWorker">Interface for Worker to be enqueued.</typeparam>
+        /// <returns>Returns true for a valid TWorker.</returns>
+        bool IsValidWorkerTypeWithImplementation<TWorker>() where TWorker : IWorker;
+
+        #endregion
```
Comment (brandongregoryscott) [inline src/AndcultureCode.CSharp.Core/Interfaces/Conductors/Jobs/IJobEnqueueConductor.cs:46]: Nitpick - matching endregion tag name

File: src/AndcultureCode.CSharp.Core/Interfaces/Workers/IWorker.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
+using System.Threading;
+using AndcultureCode.CSharp.Core.Interfaces;
+using AndcultureCode.CSharp.Core.Models.Entities.Jobs;
+
+namespace AndcultureCode.GB.Business.Core.Interfaces.Workers
```
Comment (brandongregoryscott) [inline src/AndcultureCode.CSharp.Core/Interfaces/Workers/IWorker.cs:5]: Namespace here is off - should be `AndcultureCode.CSharp.Core.Interfaces.Workers`

File: src/AndcultureCode.CSharp.Core/Interfaces/Workers/IWorker.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    {
+        #region Properties
+
+        /// <summary>
+        /// Unique name to reference worker.
+        /// </summary>
+        /// <value></value>
+        string Name { get; }
+
+        #endregion
```
Comment (brandongregoryscott) [inline src/AndcultureCode.CSharp.Core/Interfaces/Workers/IWorker.cs:20]: Nitpick - matching endregion tag names here and on line 40

File: src/AndcultureCode.CSharp.Core/Models/Entities/Jobs/Job.cs [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+
+                    workerArgsMeta.Add(meta);
+
+                    continue;
+                }
+
+                if (e?.GetType().Namespace != "System")
+                {
+                    throw new ArgumentException(
+                        "WorkerArgs must be build-in C# types",
```
Comment (brandongregoryscott) [inline src/AndcultureCode.CSharp.Core/Models/Entities/Jobs/Job.cs:144]: Nitpick - this should probably read 'built-in'

File: src/AndcultureCode.CSharp.Core/Models/Entities/Jobs/Job.cs [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+        /// </summary>
+        public void SetToInProgress()
+        {
+            if (!IsInitialState())
+            {
+                throw new Exception("The Job must be in the initial state to be set to in progress.");
+            }
+
+            // Set state.
+            StartedOn = DateTimeOffset.Now;
```
Comment (brandongregoryscott) [inline src/AndcultureCode.CSharp.Core/Models/Entities/Jobs/Job.cs:205]: Nitpick - we should probably use `UtcNow` for consistency

File: src/AndcultureCode.CSharp.Core/Models/Entities/Jobs/Job.cs [hunk 6]
```diff
@@ -0,0 +1,324 @@
...
+
+        #region Private
+
+        private struct WorkerArgsMeta
+        {
+            public object Data;
+            public string Type;
+        };
+
+        #endregion
```
Comment (brandongregoryscott) [inline src/AndcultureCode.CSharp.Core/Models/Entities/Jobs/Job.cs:322]: Nitpick - matching endregion tag name

File: test/AndcultureCode.CSharp.Core.Tests/Models/Entities/Jobs/JobTest.cs [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+                () => argList.ShouldBeOfType<List<object>>(),
+                () => argList.ShouldBeEmpty()
+            );
+        }
+
+        [Fact]
+        public void DeserializeArgs_Job_With_Malformed_Json_WorkerArgs_Throws_Error()
+        {
+            // Arrange
+            var job = new Job { WorkerArgs = "\\ Manformed Json \\" };
```
Comment (brandongregoryscott) [inline test/AndcultureCode.CSharp.Core.Tests/Models/Entities/Jobs/JobTest.cs:44]: Malformed? 😂

**Review by brandongregoryscott — COMMENTED** [review]
Some minor cleanup, @snsavage, but looks good otherwise 👍 Thanks for porting this over!

**Comment by snsavage** [comment]
@brandongregoryscott I'm working on adding to the test coverage.  I'll also fix the other issues.

**Comment by wintondeshong** [comment]
@all-contributors please add @snsavage for code, tests and documentation

**Pull request closed without merging**